contracts/Comptroller.sol

```diff
@@ -80,6 +80,22 @@ contract Comptroller is ComptrollerV5Storage, ComptrollerInterface, ComptrollerE
         admin = msg.sender;
     }
 
+    /*
+    My changes are here:
+    
+    */
+
+    function _setCollateralPaused(CToken cToken, bool state) public returns (bool){
+        
+        require(state == true || msg.sender==admin,"only admin can unpause");
+        require(msg.sender == pauseGuardian || msg.sender == admin, "only pause guardian and admin can pause");
```

Comment:
Order of these checks are switched, which means `pauseGuardian` can never pause the collateral.